# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/signaling/src/peerconnection/TransceiverImpl.cpp
# Commit: 551533329f3f
# Full Hash: 551533329f3fdb48cb2df554f2fbbb4d6ce1f877
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2018-11-09 10:01:40
# Regressor Bug: 1376873
# File Overlap Count: 3
# Description:
#   Bug 1376873 - Use audio/video sync groups; r=bwc
#   
#   This uses sync groups in the receive stream configs for the conduits rather
#   than establishing sync through direct calls. When the streams are created
#   in call.cc, ConfigureSync is called, which results in SetSync being called
# ==============================================================================

diff -r b55b0368d9f2 -r 551533329f3f media/webrtc/signaling/src/peerconnection/TransceiverImpl.cpp
--- a/media/webrtc/signaling/src/peerconnection/TransceiverImpl.cpp	Wed Feb 07 15:00:17 2018 -0500
+++ b/media/webrtc/signaling/src/peerconnection/TransceiverImpl.cpp	Fri Apr 27 07:25:19 2018 -0400
@@ -287,6 +287,12 @@
   return NS_OK;
 }
 
+void
+TransceiverImpl::ResetSync()
+{
+  mConduit->SetSyncGroup("");
+}
+
 nsresult
 TransceiverImpl::SyncWithMatchingVideoConduits(
     std::vector<RefPtr<TransceiverImpl>>& transceivers)
@@ -319,15 +325,18 @@
          transceiver->mJsepTransceiver->mRecvTrack.GetStreamIds()) {
       if (myReceiveStreamIds.count(streamId)) {
         // Ok, we have one video, one non-video - cross the streams!
-        WebrtcAudioConduit *audio_conduit =
-          static_cast<WebrtcAudioConduit*>(mConduit.get());
-        WebrtcVideoConduit *video_conduit =
-          static_cast<WebrtcVideoConduit*>(transceiver->mConduit.get());
+        mConduit->SetSyncGroup(streamId);
+        transceiver->mConduit->SetSyncGroup(streamId);
 
-        video_conduit->SyncTo(audio_conduit);
         MOZ_MTLOG(ML_DEBUG, mPCHandle << "[" << mMid << "]: " << __FUNCTION__ <<
-                            " Syncing " << video_conduit << " to "
-                            << audio_conduit);
+                            " Syncing " << mConduit.get() << " to "
+                            << transceiver->mConduit.get());
+
+        // The sync code in call.cc only permits sync between audio stream and
+        // one video stream. They take the first match, so there's no point in
+        // continuing here. If we want to change the default, we should sort
+        // video streams here and only call SetSyncGroup on the chosen stream.
+        break;
       }
     }
   }